packages/block-editor/src/components/inserter/search-results.native.js

```diff
@@ -12,31 +12,22 @@ import InserterNoResults from './no-results';
 import { store as blockEditorStore } from '../../store';
 import useBlockTypeImpressions from './hooks/use-block-type-impressions';
 
-const NON_BLOCK_CATEGORIES = [ 'reusable' ];
-const ALLOWED_EMBED_VARIATIONS = [ 'core/embed' ];
-
 function InserterSearchResults( {
 	filterValue,
 	onSelect,
 	listProps,
 	rootClientId,
 	isFullScreen,
+	allowedBlockFilter = () => true,
 } ) {
 	const { blockTypes } = useSelect(
 		( select ) => {
 			const allItems = select( blockEditorStore ).getInserterItems(
 				rootClientId
 			);
 
-			const blockItems = allItems.filter(
-				( { id, category } ) =>
-					! NON_BLOCK_CATEGORIES.includes( category ) &&
-					// We don't want to show all possible embed variations
-					// as different blocks in the inserter. We'll only show a
-					// few popular ones.
-					( category !== 'embed' ||
-						( category === 'embed' &&
-							ALLOWED_EMBED_VARIATIONS.includes( id ) ) )
+			const blockItems = allItems.filter( ( block ) =>
+				allowedBlockFilter( block, { allowReusable: true } )
```

Comment:
`allowResuable` is here to allow for searching for reusable blocks. This does make the reusable blocks appear at the bottom of the search results when the search form is focused but with an empty query.